libs/content/workspace/src/syncing.rs

```diff
@@ -3,78 +3,65 @@ use lb_rs::model::work_unit::WorkUnit;
-    // general that would allow workspace to determine if a sync is needed
-    pub fn perform_sync(&mut self) {
-        if self.status.sync_started.is_some() {
-            return;
-        }
-
-        self.status.error = None;
-        self.out.status_updated = true;
-        self.status.sync_started = Some(Instant::now());
-
-        self.tasks.queue_sync();
-    }
-
     pub fn sync_done(&mut self, outcome: CompletedSync) {
-        let CompletedSync { status_result, timing: _ } = outcome;
+        let CompletedSync { status_result, timing } = outcome;
 
         self.out.status_updated = true;
-        self.status.sync_started = None;
-        self.last_sync = Some(Instant::now());
+        self.last_sync_completed = Some(timing.completed_at);
         match status_result {
             Ok(done) => {
-                self.status.error = None;
-                self.status.offline = false;
-                self.refresh_sync_status();
+                self.status = Default::default();
+                self.tasks.queue_sync_status_update();
                 self.refresh_files(&done);
                 self.out.sync_done = Some(done)
             }
             Err(err) => match err.kind {
                 LbErrKind::ServerUnreachable => self.status.offline = true,
                 LbErrKind::ClientUpdateRequired => self.status.update_req = true,
                 LbErrKind::UsageIsOverDataCap => self.status.out_of_space = true,
-                LbErrKind::Unexpected(msg) => self.out.error = Some(msg),
+                LbErrKind::Unexpected(msg) => self.status.error = Some(msg),
                 _ => {
                     error!("Unhandled sync error: {:?}", err);
-                    self.out.error = format!("{:?}", err).into();
+                    self.status.error = format!("{:?}", err).into();
                 }
             },
         }
     }
 
-    pub fn refresh_sync_status(&mut self) {
-        let last_synced = self.core.get_last_synced_human_string().unwrap();
-        let dirty_files = self.core.get_local_changes().unwrap();
-        let pending_shares = self.core.get_pending_shares().unwrap();
-
-        let dirty = DirtynessMsg { last_synced, dirty_files, pending_shares };
-
-        self.out.status_updated = true;
-        self.status.dirtyness = dirty;
-
-        self.last_sync_status_refresh = Some(Instant::now());
-    }
-
-    pub fn refresh_files(&mut self, work: &SyncStatus) {
+    fn refresh_files(&mut self, work: &SyncStatus) {
         let server_ids = work.work_units.iter().filter_map(|wu| match wu {
             WorkUnit::LocalChange { .. } => None,
             WorkUnit::ServerChange(id) => Some(*id),
         });
 
         for id in server_ids {
             for i in 0..self.tabs.len() {
-                if self.tabs[i].id == id {
+                if self.tabs[i].id == id && !self.tabs[i].is_closing {
+                    tracing::debug!("Reloading file after sync: {}", id);
```

Comment:
nit (with `use tracing::*` for uniformness)
```suggestion
                    debug!("Reloading file after sync: {}", id);
```